vdom/schemas/vdom_schema_v0.json

```diff
@@ -1,4 +1,49 @@
 {
   "type": "object",
-  "$schema": "http://json-schema.org/draft-04/schema#"
-}
+  "$schema": "http://json-schema.org/draft-04/schema#",
+  "additionalProperties": false,
+  "required": ["tagName", "children", "attributes"],
+  "properties": {
+    "tagName": {
+      "type": "string" 
+    },
+    "children": {
+      "type": "array",
+      "items": {"$ref": "#/definitions/vdomNode"}
+    },
+    "attributes": {
+      "type": "object"   
+    },
+    "key": {
+      "oneOf": [{"type": "number"}, {"type": "string"}, {"type": "null"}]
+    }
+  },
+  "definitions": {
+    "vdomEl": {
+      "type": "object",
+      "required": ["tagName", "children", "attributes"],
+      "properties": {
+        "tagName": {
+          "type": "string" 
+        },
+        "children": {
+          "type": "array",
+          "items": {"$ref": "#/definitions/vdomNode"}
+        },
+        "attributes": {
+          "type": "object"   
+        },
+        "key": {
+          "oneOf": [{"type": "number"}, {"type": "string"}, {"type": "null"}]
+        }
+      }
+    },
+    "vdomNode": {
+      "oneOf": [
+        {"type": "object"},
+        {"type": "array", "items": {"$ref": "#/definitions/vdomNode"}},
+        {"type": "null"}
+      ]
```

Comment:
The full vdom node definition should be exactly this:

```
    "vdomNode": {
      "oneOf": [
        { "type": "array", "items": { "$ref": "#/definitions/vdomNode" } },
        { "$ref": "#/definitions/vdomEl" },
        { "type": "string" },
        { "type": "null" }
      ]
    }
```

Reply:
This is what I was trying to do, which threw a ValidationError. I set it to {"type": "object"} just to get it to render.

I'll debug more.